### ISSUE TITLE
When resubscribing, the previous subscription shipping address is ignored.

### ISSUE THREAD

**Issue by Jinksi** [Automattic/woocommerce-payments#5293]
### Describe the bug

If a customer is resubscribing to an existing subscription with a shipping address different to their billing address, the checkout "Ship to a different address" checkbox should be checked.

### To Reproduce

**Resubscribe**
1. Purchase a subscription product as a customer, **using a different shipping address to billing address**.
1. Clear the shopper's cart.
1. Cancel the subscription in `wp-admin` as the merchant.
1. As the shopper, visit **My Account**, find the active subscription and click "**Resubscribe**".
1. Notice at checkout that the "Ship to a different address" checkbox has **not** been checked.

### Expected behaviour

If a customer is resubscribing to an existing subscription with a shipping address different to their billing address, the checkout "Ship to a different address" checkbox should be checked.

<img src="https://user-images.githubusercontent.com/3147296/172106528-2cc560d8-9cff-479a-9494-3da078726bcd.png" alt="Ship to different address checkbox checked" width="500" />

### Actual behaviour

The checkout "Ship to a different address" checkbox is not checked when resubscribing to an existing subscription with a shipping address different to billing address,

### Additional context

* subscriptions-core issue: https://github.com/Automattic/woocommerce-subscriptions-core/issues/178
* WCS issue: 4333-gh-woocommerce/woocommerce-subscriptions

**Comment by haszari** [comment]
Marking as low priority for now since this is a resubscribe flow (likely low impact).

**Comment by Jinksi** [comment]
This one has a [subs-core PR](https://github.com/Automattic/woocommerce-subscriptions-core/pull/179) ready for review 😆 (I added it to In Progress on the maintenance board)

**Comment by haszari** [comment]
> This one has a https://github.com/Automattic/woocommerce-subscriptions-core/pull/179 ready for review 😆 (I added it to In Progress on the maintenance board)

Yikes, that's been sitting there a while - thanks for digging it up! @Jinksi can you a first pass and update the branch / conflicts to unblock a review? If it's risky feel free to put on the backburner explicitly, 🤞 but would be awesome to get a fix shipped!

**Comment by Jinksi** [comment]
> can you a first pass and update the branch / conflicts to unblock a review?

Done 👍

This one is low-risk IMO, checks the "Use a different shipping address" checkout checkbox for switches and resubscribes when appropriate, where it was only working for renewals previously.

**Comment by RadoslavGeorgiev** [comment]
@haszari should we close this WCPay issue in favor of [the subscriptions-core PR](https://github.com/Automattic/woocommerce-subscriptions-core/pull/179)? It seems like WCPay is not involved here.

**Comment by haszari** [comment]
> should we close this WCPay issue in favor of https://github.com/Automattic/woocommerce-subscriptions-core/pull/179? It seems like WCPay is not involved here

No - we prefer to have an issue logged against the product that's affected. Subscriptions core is a library. Interested WCPay merchants can follow this issue to track when it's fixed, etc.

This issue will be closed when a subs-core version with a fix is merged into WCPay code base.

**Comment by Jinksi** [comment]
Fixed in subscriptions-core 5.3.0 #5469
